web-app/js/Habitat.js

```diff
@@ -289,20 +290,27 @@ var Habitat = function (c) {
         };
 
         self.updateChart = function(habitat, list){
-            var data = {
-                speciesList: JSON.stringify(list),
-                config: habitat.name(),
-                biocacheServiceUrl: pj.records.getDataresource().biocacheServiceUrl
-            };
-            if(config.doSaveQuery){
-                self.saveQuery(data).then(function(qid){
-                    var data = {q:"qid:"+qid, biocacheServiceUrl:records.getDataresource().biocacheServiceUrl};
+            if (config.records.getDataresource() !== undefined) {
+                var data = {
+                    speciesList: JSON.stringify(list),
+                    config: habitat.name(),
+                    biocacheServiceUrl: config.records.getDataresource().biocacheServiceUrl
+                };
+                if (config.doSaveQuery) {
+
+                    self.saveQuery(data).then(function (qid) {
```

Comment:
move this save query to PJ.js. Save query should only run in two conditions, 1. when a node is clicked. 2. when source data is changed. Here, it looks like it is running every time updatechart is called.